sort_names.py

```diff
@@ -1,95 +1,163 @@
-import csv
+import csv  # work with .csv files
 import sys
 import time
+import urllib.request
+from collections import defaultdict
 from fileinput import filename
-from deepface import DeepFace
 
+# modules
+import filters
+import inputs
 
 filename = sys.argv[1]
 
 # ---- INPUT PARAMETERS ----
-limit_age = 25  # not younger than
-races = [
-    "black",
-    "white",
-    "asian",
-]  # enter necessary races. List of possible races - ['asian', 'indian', 'black', 'white', 'middle eastern', 'latino hispanic']
-forbidden_word = "NFT"  # a forbidden word in the current position of person
-black_list = "Indusi.csv"  # Filter by name. CSV file with two columns - First_name and Second_name
+LIMIT_AGE = 25  # not younger than
+# enter necessary races. List of possible races - ['asian', 'indian', 'black', 'white', 'middle eastern', 'latino hispanic']
+races = ["asian", "black", "white"]
+DEL_PEOPLE_WITHOUT_AVATAR = False  # past 'True' or 'False'
 
 
-def get_short_list(filename, first_names, last_names):
+def conditions(row, dict_races, first_names, last_names, positions):
+    """Check all conditions and if row is correct - return True, else - False.
+    At the same time returns a tuple with counters
+    """
+    counts = []
+    conditions = [
```

Comment:
If it makes sense: lets convert it to dict: {condition: function_name (function that should be executed, when condition is satisfied or not satisfied)}

Reply:
Sorry, let me write the prototype.

Reply:
```
    counts = []

    function filter_face():
        age, race, curent_race = filters.face_filter(row[30], LIMIT_AGE, races)
        if curent_race is not None:
            if age:
                counts.append("count_age")
                dict_races[curent_race] += 1
                if race:
                    counts.extend(["count", "count_race"])
                    return True, counts
                return False, counts
            return False, counts
        counts.extend(["count_unreadable_ava", "count"])
        return True, counts

    conditions1 = [
        filters.name_filter(row[24], first_names),
        row[25] not in last_names,
        filters.position_filter(row[43], positions),
    ]
    functions1 = [
        lambda: counts.append("count_name"),
        lambda: counts.append("count_last"),
        lambda: counts.append("count_pos"),
    ]
    returns1 = [
        (False, counts),
        (False, counts),
        (False, counts),
    ]

    for cond, func, ret in zip(conditions1, functions1, returns1):
        if cond:
            func()
        else:
            return ret

    conditions2 = [
        (row[30] == "") and DEL_PEOPLE_WITHOUT_AVATAR,
        (row[30] == "") and not DEL_PEOPLE_WITHOUT_AVATAR,
    ]

    functions2 = [
        lambda: counts.append("count_no_link"),
        lambda: counts.extend(["count", "count_no_link"]),
    ]

    returns2 = [
        (False, count),
        (True, count),
    ]
    
    for cond, func, ret in zip(conditions2, functions2, returns2):
        if cond:
            func()
            return 

    return filter_face()
```

Reply:
```
def conditions(row, dict_races, first_names, last_names, positions):
    """Check all conditions and if row is correct - return True, else - False.
    At the same time returns a tuple with counters
    """
    counts = []

    function filter_face():
        age, race, curent_race = filters.face_filter(row[30], LIMIT_AGE, races)
        if curent_race is not None:
            if age:
                counts.append("count_age")
                dict_races[curent_race] += 1
                if race:
                    counts.extend(["count", "count_race"])
                    return True, counts
                return False, counts
            return False, counts
        counts.extend(["count_unreadable_ava", "count"])
        return True, counts

    conditions1 = [
        filters.name_filter(row[24], first_names),
        row[25] not in last_names,
        filters.position_filter(row[43], positions),
    ]
    functions1 = [
        lambda: counts.append("count_name"),
        lambda: counts.append("count_last"),
        lambda: counts.append("count_pos"),
    ]
    returns1 = [
        (False, counts),
        (False, counts),
        (False, counts),
    ]

    for cond, func, ret in zip(conditions1, functions1, returns1):
        if cond:
            func()
        else:
            return ret

    conditions2 = [
        (row[30] == "") and DEL_PEOPLE_WITHOUT_AVATAR,
        (row[30] == "") and not DEL_PEOPLE_WITHOUT_AVATAR,
    ]

    functions2 = [
        lambda: counts.append("count_no_link"),
        lambda: counts.extend(["count", "count_no_link"]),
    ]

    returns2 = [
        (False, count),
        (True, count),
    ]
    
    for cond, func, ret in zip(conditions2, functions2, returns2):
        if cond:
            func()
            return ret

    return filter_face()
```

Reply:
there seems to be a new working version, but not sure if it's any more readable. I push it into the next version 1.5